lib/common/property-utils.ts

```diff
@@ -0,0 +1,6 @@
+export function defineProperty(target: Object, propertyName: string, getter: () => any, setter?: (value: any) => void) {
```

Comment:
This is less type-safe than the original approach, I'm not personally sure we need this helper as it just introduces a level of abstraction that feels not-needed.  If we do want it though this should definitely by typed like so.

```suggestion
export function defineProperty<T extends object, K extends keyof T>(target: T, propertyName: K, getter: () => T[K], setter?: (value: T[K]) => void) {
```